[[ Bug 19474 ]] Improve script object description

runrevmark opened the pull request:
This patch changes the description of a script object from
`<deleted object>` and `<script object>` to `<deleted script object>`
and `<script object OBJECT_LONG_NAME>`.

runrevmark committed:
[[ Bug 19474 ]] Improve script object description

This patch changes the description of a script object from
<deleted object> and <script object> to <deleted script object>
and <script object OBJECT_LONG_NAME>.

peter-b commented:
@runrevmark Does this need tests and/or a release note?

runrevmark commented:
It's an internal change to the description format - I don't think it strictly needs either right now :)

peter-b commented on engine/src/module-engine.cpp:
```diff
@@ -674,8 +674,25 @@ static hash_t __MCScriptObjectHash(MCValueRef p_value)
-    r_description = MCSTR("<script object>");
-    return true;
+    __MCScriptObjectImpl *self;
+    self = (__MCScriptObjectImpl *)MCValueGetExtraBytesPtr(p_value);
```
Per the style guide, please use RAII and C++-style casts, i.e.

```c++
auto self = reinterpret_cast<__MCScriptObjectImpl*>(MCValueGetExtraBytesPtr(p_value));
```

We allow `auto` here because the type is explicit.

runrevmark committed:
[[ Bug 19474 ]] Update to conform to coding standards

This patch uses 'auto' and reinterpret_cast to fetch
the Impl from the script object.

peter-b commented:
@livecode-vulcan review ok 5aadb12

livecode-vulcan commented:
💙 review by @peter-b ok 5aadb12c0e6fe186011d666f12f0bead45f2c203

livecode-vulcan commented:
😎 test success 5aadb12c0e6fe186011d666f12f0bead45f2c203

* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/7)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/7)
* try-community-js-emscripten: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten/builds/14)
* try-community-universal-ios-iphoneos10.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.2/builds/12)
* try-community-universal-ios-iphonesimulator10.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.2/builds/12)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/6)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/8)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/10)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/12)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/10)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/12)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/12)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/11)
